app/models/organization.rb

```diff
@@ -158,13 +158,16 @@ def law_enforcement_missing_verified_features?
   end
 
   def paid_for?(feature_name)
-    return false unless feature_name.present? && paid_feature_slugs.is_a?(Array)
-    # If kinds is an array, make sure they all are permitted kinds
-    if feature_name.is_a?(Array)
-      return false unless feature_name.any? # If they passed an empty array, it's false
-      return feature_name.none? { |k| !paid_for?(k) }
+    features =
+      Array(feature_name)
+        .map { |name| name.strip.downcase.gsub(/\s/, "_") }
+
+    return false unless features.present? && paid_feature_slugs.is_a?(Array)
```

Comment:
Organizations use this to set their paid_feature_slugs:

```ruby
self.paid_feature_slugs = current_invoices.feature_slugs
```

Which I think results in it being assigned to `nil` - and ambassador organizations won't have active invoices, so this will probably result in the ambassador organization not getting access to `unstolen_notifications`. I think it would be fine to switch the above assignment to fallback to assigning to an empty array.

Reply:
Hmmmmm. But it does pass the spec, so maybe this isn't necessary?

Reply:
It's an AR relation, so it ends up evaluating to an array:

```rb
Organization
  .ambassador
  .limit(1)
  .map{ |o| [o.current_invoices.feature_slugs, o.paid_feature_slugs] }

  Organization Load (0.6ms)  SELECT  "organizations".* FROM "organizations" WHERE "organizations"."deleted_at" IS NULL AND "organizations"."kind" = $1  ORDER BY "organizations"."name" ASC LIMIT 1  [["kind", 8]]
  Invoice Load (0.2ms)  SELECT "invoices".* FROM "invoices" WHERE "invoices"."organization_id" = $1 AND "invoices"."is_active" = $2  [["organization_id", 899], ["is_active", "t"]]

[
  [0] [
    [0] [],
    [1] []
  ]
]

```

Interestingly, it looks like AR internally consolidates the N+1 query from `Invoice.feature_slugs`. It still shows up when not invoking it through an association though, so I'll refactor while I'm in here. 

```rb
2.5.1 (bikeindex)[43] » Invoice.feature_slugs

  Invoice Load (0.2ms)  SELECT "invoices".* FROM "invoices"
   (0.2ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1038]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1047]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1040]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1048]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1030]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1031]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1036]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1049]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1041]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1044]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1051]]
   (0.4ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1046]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1052]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1045]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1042]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1023]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1024]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1029]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1028]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1039]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1001]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1025]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1035]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 99]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 1037]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 100]]
   (0.1ms)  SELECT "paid_features"."feature_slugs" FROM "paid_features" INNER JOIN "invoice_paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id" WHERE "invoice_paid_features"."invoice_id" = $1  [["invoice_id", 101]]
[
  [ 0] "show_bulk_import",
  [ 1] "reg_secondary_serial",
...
]
```

vs

```rb
2.5.1 (bikeindex)[46] » Invoice.feature_slugs

   (0.5ms)  SELECT feature_slugs FROM "invoices" LEFT OUTER JOIN "invoice_paid_features" ON "invoice_paid_features"."invoice_id" = "invoices"."id" LEFT OUTER JOIN "paid_features" ON "paid_features"."id" = "invoice_paid_features"."paid_feature_id"
[
  [ 0] "show_partial_registrations",
  [ 1] "bike_search",
  [ 2] "csv_exports",
 . . .
]

2.5.1 (bikeindex)[47] » ss Invoice.feature_slugs

From: /Users/jmromer/Work/bike_index/app/models/invoice.rb @ line 33:
Owner: #<Class:Invoice(id: integer, organization_id: integer, first_invoice_id: integer, is_active: boolean, force_active: boolean, subscription_start_at: datetime, subscription_end_at: datetime, amount_due_cents: integer, amount_paid_cents: integer, created_at: datetime, updated_at: datetime, notes: text)>
Visibility: public
Number of lines: 4

def self.feature_slugs
  # all.map(&:feature_slugs).flatten.uniq
  includes(:paid_features).pluck(:feature_slugs).flatten.uniq
end

```